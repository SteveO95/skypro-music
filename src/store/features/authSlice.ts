import { getJWTokens, login } from "@/services/api";
import { deleteAuthCookie, setAuthCookie } from "@/services/cookie";
import { AuthType } from "@/types/auth";
import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";

type AuthStateType = {
  isAuth: boolean;
  accessToken: string | undefined;
  refreshToken: string | undefined;
  username: string | undefined;
  email: string | undefined;
  id: string | undefined;
};

export const setUserAuth = createAsyncThunk(
  "auth/setUserAuth",
  async (
    { email, password }: { email: string; password: string },
    { dispatch, rejectWithValue }
  ) => {
    try {
      const loginResult = await login({ email, password });
      const tokenResult = await getJWTokens({ email, password });
      const authData = {
        id: loginResult._id,
        email: loginResult.email,
        username: loginResult.username,
        accessToken: tokenResult.access,
        refreshToken: tokenResult.refresh,
      };
      dispatch(setAuth(authData));
      return authData;
    } catch (error) {
      if (error instanceof Error) {
        return rejectWithValue(error.message);
      } else {
        return rejectWithValue("Something went wrong");
      }
    }
  }
);

export const setUserLogout = createAsyncThunk("auth/setUserLogout", async (_, { dispatch }) => {
  dispatch(unsetAuth());
});

const initialState: AuthStateType = {
  isAuth: false,
  accessToken: undefined,
  refreshToken: undefined,
  username: undefined,
  email: undefined,
  id: undefined,
};

const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    setAuth: (state, action: PayloadAction<AuthType>) => {
      state.isAuth = true;
      state.id = action.payload.id;
      state.username = action.payload.username;
      state.email = action.payload.email;
      state.accessToken = action.payload.accessToken;
      state.refreshToken = action.payload.refreshToken;

      setAuthCookie(
        {
          id: action.payload.id,
          username: action.payload.username,
          email: action.payload.email,
          accessToken: action.payload.accessToken,
          refreshToken: action.payload.refreshToken,
        },
        { maxAge: 12 * 3600 }
      );
    },
    unsetAuth: (state) => {
      state.isAuth = initialState.isAuth;
      state.id = initialState.id;
      state.username = initialState.username;
      state.email = initialState.email;
      state.accessToken = initialState.accessToken;
      state.refreshToken = initialState.refreshToken;
      deleteAuthCookie();
    },
  },
  extraReducers(builder) {
    builder.addCase(setUserAuth.rejected, (state, action) => {});
  },
});

const { unsetAuth } = authSlice.actions;
export const { setAuth } = authSlice.actions;
export const authReducer = authSlice.reducer;
