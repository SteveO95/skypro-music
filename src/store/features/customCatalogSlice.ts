import { getCatalogs } from "@/services/api";
import { CustomCatalogType } from "@/types/customCatalog";
import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";

const imagesCompare: { [key: string]: string } = {
  ["Инди-заряд"]: "/img/playlist03.png",
  ["Танцевальные хиты"]: "/img/playlist02.png",
  ["Плейлист дня"]: "/img/playlist01.png",
};

type CustomCatalogStateType = {
  currentCatalog?: string;
  customCatalogs?: CustomCatalogType[];
};

const initialState: CustomCatalogStateType = {
  currentCatalog: undefined,
  customCatalogs: undefined,
};

export const getCustomCatalogs = createAsyncThunk(
  "customCatalog/getCustomCatalogs",
  async (_, { rejectWithValue }) => {
    try {
      const data = await getCatalogs();
      return data.map((playlist) => {
        return {
          ...playlist,
          imagePath: imagesCompare[playlist.name] || "",
        };
      });
    } catch (err) {
      return rejectWithValue("Ошибка получения списка подборки");
    }
  }
);

const customCatalogSlice = createSlice({
  name: "customCatalog",
  initialState,
  reducers: {},
  extraReducers(builder) {
    builder.addCase(getCustomCatalogs.fulfilled, (state, action) => {
      state.customCatalogs = action.payload;
    });
    builder.addCase(getCustomCatalogs.rejected, (state, action) => {
      state.customCatalogs = [];
      console.error(action.payload);
    });
  },
});

export const customCatalogReducer = customCatalogSlice.reducer;
