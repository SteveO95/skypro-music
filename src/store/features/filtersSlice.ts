import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export type FiltersState = {
  genres: string[];
  authors: string[];
  keyword: string;
  sortOption: string;
};

const initialFiltersState: FiltersState = {
  genres: [],
  authors: [],
  keyword: "",
  sortOption: "По умолчанию",
};

const filtersSlice = createSlice({
  name: "filters",
  initialState: initialFiltersState,
  reducers: {
    updateFilters(state, action: PayloadAction<Partial<FiltersState>>) {
      return {
        ...state,
        ...action.payload,
      };
    },
    resetFilters() {
      return initialFiltersState;
    },
  },
});

export const { updateFilters, resetFilters } = filtersSlice.actions;

export const filterReducer = filtersSlice.reducer;
