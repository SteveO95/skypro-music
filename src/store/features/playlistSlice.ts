import { tracksApi } from '@/api/tracksApi'
import { TrackDataType } from '@/lib/types'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'

type PlaylistStateType = {
	mainPlaylist: TrackDataType[]
	shufflePlaylist: TrackDataType[]
	favoritePlaylist: TrackDataType[]
	selectionPlaylist: TrackDataType[]
	selectionData: { name: string; items: number[] }
	currTrack: TrackDataType | null
	isLoading: boolean
	isPlaying: boolean
	isLoop: boolean
	isShuffle: boolean
	filterOptions: {
		searchValue: string
		author: string[]
		orderData: string
		genre: string[]
	}
}

const initialState: PlaylistStateType = {
	mainPlaylist: [],
	shufflePlaylist: [],
	favoritePlaylist: [],
	selectionPlaylist: [],
	selectionData: { name: '', items: [] },
	currTrack: null,
	isLoading: true,
	isPlaying: false,
	isLoop: false,
	isShuffle: false,
	filterOptions: {
		searchValue: '',
		author: [],
		orderData: 'По умолчанию',
		genre: [],
	},
}

const playlistSlice = createSlice({
	name: 'playlist',
	initialState,
	reducers: {
		setCurrTrack: (
			state,
			action: PayloadAction<{
				currTrack: TrackDataType
				mainPlaylist: TrackDataType[]
			}>,
		) => {
			const sortPlaylist = [...action.payload.mainPlaylist].sort(
				() => 0.5 - Math.random(),
			)

			state.currTrack = action.payload.currTrack
			state.mainPlaylist = action.payload.mainPlaylist
			state.shufflePlaylist = sortPlaylist
		},
		setLike: (state, action: PayloadAction<TrackDataType>) => {
			state.favoritePlaylist.push(action.payload)
		},
		setDislike: (state, action: PayloadAction<TrackDataType>) => {
			const filterNoFavorite = state.favoritePlaylist.filter(
				track => track._id !== action.payload._id,
			)

			state.favoritePlaylist = filterNoFavorite
		},
		setPrevTrack: state => {
			const playlist = state.isShuffle
				? state.shufflePlaylist
				: state.mainPlaylist
			const currIndex = playlist.findIndex(
				track => track._id === state.currTrack?._id,
			)

			if (!currIndex) {
				return
			}

			state.currTrack = playlist[currIndex - 1]
		},
		setNextTrack: state => {
			const playlist = state.isShuffle
				? state.shufflePlaylist
				: state.mainPlaylist
			const currIndex = playlist.findIndex(
				track => track._id === state.currTrack?._id,
			)

			if (currIndex === playlist.length - 1) {
				return
			}

			state.currTrack = playlist[currIndex + 1]
		},
		setIsPlaying: (state, action: PayloadAction<boolean>) => {
			state.isPlaying = action.payload
		},
		setIsLoop: (state, action: PayloadAction<boolean>) => {
			state.isLoop = action.payload
		},
		setIsShuffle: (state, action: PayloadAction<boolean>) => {
			state.isShuffle = action.payload
		},
		setSearchValue: (state, action) => {
			state.filterOptions.searchValue = action.payload
		},
		setAuthor: (state, action) => {
			const value = action.payload
			const author = state.filterOptions.author

			if (author.includes(value)) {
				state.filterOptions.author = author.filter(author => author !== value)
			} else {
				author.push(value)
			}
		},
		setOrderData: (state, action) => {
			state.filterOptions.orderData = action.payload
		},
		setGenre: (state, action) => {
			const value = action.payload
			const genre = state.filterOptions.genre

			if (genre.includes(value)) {
				state.filterOptions.genre = genre.filter(genre => genre !== value)
			} else {
				genre.push(value)
			}
		},
		resetFilterOptions: state => {
			state.filterOptions = {
				author: [],
				genre: [],
				orderData: 'По умолчанию',
				searchValue: '',
			}
		},
	},

	extraReducers: builder => {
		builder
			.addCase(tracksApi.getTracks.fulfilled, (state, action) => {
				state.mainPlaylist = action.payload
				state.isLoading = false
			})
			.addCase(tracksApi.getFavoriteTracks.fulfilled, (state, action) => {
				state.favoritePlaylist = action.payload
				state.isLoading = false
			})
			.addCase(tracksApi.getSelections.fulfilled, (state, action) => {
				const filterSelection = state.mainPlaylist.filter(track => {
					return state.selectionData?.items.includes(track._id)
				})

				state.selectionPlaylist = filterSelection
				state.selectionData = action.payload
				state.isLoading = false
			})
	},
})

export const {
	setCurrTrack,
	setLike,
	setDislike,
	setPrevTrack,
	setNextTrack,
	setIsPlaying,
	setIsLoop,
	setIsShuffle,
	setSearchValue,
	setAuthor,
	setOrderData,
	setGenre,
	resetFilterOptions,
} = playlistSlice.actions
export const playlistReducer = playlistSlice.reducer
