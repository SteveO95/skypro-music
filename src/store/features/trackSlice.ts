import { createAsyncThunk, createSlice, current, PayloadAction } from "@reduxjs/toolkit";
import { TrackType } from "@/types/tracks";
import { fetchFavoriteTracks, getAllTracks } from "@/services/api";
import { FilterType } from "@/types/filter";
import { SortType } from "@/types/sort";
import { act } from "react";

type TrackStateType = {
  initialPlaylistState?: TrackType[];
  currentPlaylistState?: TrackType[];
  likedPlaylistState: TrackType[];
  currentTrackState: TrackType | undefined;
  currentPlaylistTypeState: "All" | "Favorites" | "Сollection";
  isPlayingState: boolean;
  isShuffleState: boolean;
  filters: FilterType[];
  sort: Partial<Record<SortType["type"], SortType>>;
};

const initialState: TrackStateType = {
  initialPlaylistState: undefined,
  currentPlaylistState: undefined,
  likedPlaylistState: [],
  currentTrackState: undefined,
  currentPlaylistTypeState: "All",
  isPlayingState: false,
  isShuffleState: false,
  filters: [],
  sort: {},
};

export const getInitialPlaylist = createAsyncThunk<TrackType[], void, { rejectValue: string }>(
  "playlist/getInitialPlaylist",
  async (_, { rejectWithValue }) => {
    try {
      const data = await getAllTracks();
      const tracks = data.map((track: { _id: string; [key: string]: any }) => {
        const { _id, ...rest } = track;
        return { id: _id, ...rest };
      });
      return tracks;
    } catch (err) {
      return rejectWithValue("Произошла ошибка при получении списка треков");
    }
  }
);

export const getFavoriteTrack = createAsyncThunk(
  "playlist/getFavoriteTracks",
  async (
    { accessToken, refreshToken }: { accessToken: string; refreshToken: string },
    { rejectWithValue }
  ) => {
    try {
      const data = await fetchFavoriteTracks({ accessToken, refreshToken });

      const tracks = data.data.map((track: { _id: string; [key: string]: any }) => {
        const { _id, ...rest } = track;
        return { id: _id, ...rest };
      });
      return tracks;
    } catch (err) {
      return rejectWithValue("Произошла ошибка при получении любимых треков");
    }
  }
);

const filterPlaylistState = (
  initialPlaylistState: TrackStateType["initialPlaylistState"],
  filters: TrackStateType["filters"]
): TrackStateType["currentPlaylistState"] => {
  return initialPlaylistState?.filter((track) => {
    if (filters.length <= 0) return true;

    const filtersByType = filters.reduce<Record<string, string[]>>((acc, filter) => {
      if (!acc[filter.type]) {
        acc[filter.type] = [];
      }
      acc[filter.type].push(filter.value);
      return acc;
    }, {});

    return Object.entries(filtersByType).every(([type, values]) => {
      if (type === "search") {
        return (
          values.some((value) => track.name.toLowerCase().includes(value.toLowerCase())) ||
          values.some((value) => track.author.toLowerCase().includes(value.toLocaleLowerCase()))
        );
      }

      if (type === "genre") {
        return values.some((value) => track.genre.includes(value));
      }

      return values.some((value) => track[type as keyof typeof track] === value);
    });
  });
};

const trackSlice = createSlice({
  name: "track",
  initialState,
  reducers: {
    setPlaylistType: (state, action: PayloadAction<"All" | "Favorites" | "Сollection">) => {
      state.currentPlaylistTypeState = action.payload;
    },
    setTrack: (state, action: PayloadAction<TrackType | undefined>) => {
      state.currentTrackState = action.payload;
    },
    setInitialPlaylist: (state, action: PayloadAction<TrackType[]>) => {
      state.currentPlaylistState = action.payload;
      state.initialPlaylistState = action.payload;
    },
    setPreviousTrack: (state) => {
      if (state.currentPlaylistState === undefined) return;

      const currentTrackIndex = state.currentPlaylistState.findIndex(
        (track) => track.id === state.currentTrackState?.id
      );
      if (currentTrackIndex === -1 || currentTrackIndex <= 0) return;
      state.currentTrackState = state.currentPlaylistState[currentTrackIndex - 1];
      state.isPlayingState = true;
    },
    setNextTrack: (state) => {
      if (state.currentPlaylistState === undefined) return;

      const currentTrackIndex = state.currentPlaylistState.findIndex(
        (track) => track.id === state.currentTrackState?.id
      );

      if (currentTrackIndex === -1 || currentTrackIndex + 1 >= state.currentPlaylistState.length)
        return;

      state.currentTrackState = state.currentPlaylistState[currentTrackIndex + 1];
      state.isPlayingState = true;
    },
    setPlaying: (state, action: PayloadAction<boolean>) => {
      state.isPlayingState = action.payload;
    },
    setShufflePlaylist: (state, action: PayloadAction<boolean>) => {
      if (state.initialPlaylistState === undefined) return;

      state.currentPlaylistState = !action.payload
        ? state.initialPlaylistState
        : state.initialPlaylistState.toSorted((a, b) => Math.random() - 0.5);

      state.isShuffleState = action.payload;
    },
    setDislikeTrack: (state, action: PayloadAction<number>) => {
      const trackID = action.payload;
      state.likedPlaylistState = state.likedPlaylistState.filter((track) => track.id !== trackID);

      if (state.currentPlaylistTypeState === "Favorites") {
        if (state.currentPlaylistState !== undefined) {
          state.currentPlaylistState = state.currentPlaylistState.filter(
            (track) => track.id !== trackID
          );
        }
        if (state.initialPlaylistState !== undefined) {
          state.initialPlaylistState = state.initialPlaylistState.filter(
            (track) => track.id !== trackID
          );
        }
      }
    },
    setLikeTrack: (state, action: PayloadAction<number>) => {
      if (state.currentPlaylistState === undefined) return;
      const trackID = action.payload;
      const track = state.currentPlaylistState.find((track) => track.id === trackID);
      if (!track) return;
      state.likedPlaylistState.push(track);
    },
    setLikedPlaylist: (state, action: PayloadAction<TrackType[]>) => {
      state.likedPlaylistState = action.payload;
    },
    filterPlaylist: (
      state,
      action: PayloadAction<{ operation: "add" | "delete"; filter: FilterType }>
    ) => {
      const { operation, filter } = action.payload;

      if (operation === "add") {
        if (filter.type === "search") {
          state.filters = state.filters.filter((f) => f.type !== "search");
          state.filters.push(filter);
        } else {
          state.filters.push(filter);
        }
      } else {
        state.filters = state.filters.filter(
          (f) => f.type !== filter.type || f.value !== filter.value
        );
      }

      state.currentPlaylistState = filterPlaylistState(state.initialPlaylistState, state.filters);
    },
    sortPlaylist: (state, action: PayloadAction<SortType>) => {
      state.sort[action.payload.type] = action.payload;

      if (!state.currentPlaylistState) return;

      if (action.payload.direction === "default") {
        state.currentPlaylistState = filterPlaylistState(state.initialPlaylistState, state.filters);
        return;
      }

      state.currentPlaylistState = [...state.currentPlaylistState].sort((a, b) => {
        const dateA = new Date(a.release_date).getTime();
        const dateB = new Date(b.release_date).getTime();

        return action.payload.direction === "asc" ? dateB - dateA : dateA - dateB;
      });
    },
  },
  extraReducers: (builder) => {
    builder.addCase(getInitialPlaylist.fulfilled, (state, action) => {
      state.initialPlaylistState = action.payload;
      state.currentPlaylistState = action.payload;
    });
    builder.addCase(getInitialPlaylist.rejected, (state, action) => {
      state.initialPlaylistState = [];
      state.currentPlaylistState = [];
      console.error(action.payload);
    });

    builder.addCase(getFavoriteTrack.fulfilled, (state, action) => {
      state.likedPlaylistState = action.payload;
    });
    builder.addCase(getFavoriteTrack.rejected, (state, action) => {
      console.error(action.payload);
    });
  },
});

export const {
  setPlaylistType,
  setTrack,
  setNextTrack,
  setPreviousTrack,
  setPlaying,
  setInitialPlaylist,
  setShufflePlaylist,
  setDislikeTrack,
  setLikeTrack,
  setLikedPlaylist,
  filterPlaylist,
  sortPlaylist,
} = trackSlice.actions;
export const trackReducer = trackSlice.reducer;
