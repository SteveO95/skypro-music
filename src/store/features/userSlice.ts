import { userApi } from '@/api/userApi'
import { TokenType, UserDataType } from '@/lib/types'
import { getTokensFromLs, getUserFromLs } from '@/utils/helpers'
import { createSlice } from '@reduxjs/toolkit'

type UserStateType = {
	user: UserDataType | null
	tokens: TokenType | null
}

const initialState: UserStateType = {
	user: getUserFromLs(),
	tokens: getTokensFromLs(),
}

const userSlice = createSlice({
	name: 'user',
	initialState,
	reducers: {
		logout: state => {
			state.user = null
			localStorage.removeItem('user')
			localStorage.removeItem('tokens')
		},
	},

	extraReducers: builder => {
		builder
			.addCase(userApi.getUser.fulfilled, (state, action) => {
				state.user = action.payload
				localStorage.setItem('user', JSON.stringify(state.user))
			})
			.addCase(userApi.regUser.fulfilled, (state, action) => {
				state.user = action.payload
			})
			.addCase(userApi.getToken.fulfilled, (state, action) => {
				state.tokens = action.payload
				localStorage.setItem('tokens', JSON.stringify(state.tokens))
			})
	},
})

export const { logout } = userSlice.actions
export const userReducer = userSlice.reducer
