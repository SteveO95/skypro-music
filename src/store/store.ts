import {
	TypedUseSelectorHook,
	useDispatch,
	useSelector,
	useStore,
} from 'react-redux'
import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { userReducer } from './features/userSlice'
import { playlistReducer } from './features/playlistSlice'

export const makeStore = () => {
	return configureStore({
		reducer: combineReducers({
			user: userReducer,
			playlist: playlistReducer,
		}),
	})
}

export type AppStore = ReturnType<typeof makeStore>
export type RootState = ReturnType<AppStore['getState']>
export type AppDispatch = AppStore['dispatch']

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppStore: () => AppStore = useStore
