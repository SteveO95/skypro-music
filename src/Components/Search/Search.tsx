import React, { ChangeEvent } from 'react'
import styles from './Search.module.css'
import { useAppDispatch, useAppSelector } from '@/store/store'
import { setSearchValue } from '@/store/features/playlistSlice'

export default function Search() {
	const dispatch = useAppDispatch()
	const { filterOptions } = useAppSelector(state => state.playlist)

	const handleSearch = (event: ChangeEvent<HTMLInputElement>) => {
		dispatch(setSearchValue(event.target.value))
	}

	return (
		<div className={styles.centerblock__search}>
			<svg className={styles.search__svg}>
				<use xlinkHref='/img/icon/sprite.svg#icon-search' />
			</svg>
			<input
				className={styles.search__text}
				type='search'
				placeholder='Поиск'
				name='search'
				value={filterOptions.searchValue}
				onChange={handleSearch}
			/>
		</div>
	)
}
