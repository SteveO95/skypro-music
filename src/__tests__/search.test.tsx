import Search from '@/components/Search/Search'
import { useAppDispatch, useAppSelector } from '@/store/store'
import { fireEvent, render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { setSearchValue } from '@/store/features/playlistSlice'

jest.mock('../store/store.ts')

const mockUseAppDispatch = useAppDispatch as jest.Mock
const mockUseAppSelector = useAppSelector as jest.Mock
const mockDispatch = jest.fn()

describe('Поиск', () => {
	beforeEach(() => {
		jest.clearAllMocks()
		mockUseAppDispatch.mockReturnValue(mockDispatch)
	})
	afterEach(() => {
		cleanup()
	})

	const searchInputTests = [
		{
			searchValue: '',
		},
		{
			searchValue: '1',
		},
		{
			searchValue: 'один',
		},
	]

	searchInputTests.forEach(value => {
		it('рендер компонента', () => {
			mockUseAppSelector.mockReturnValue({
				filterOptions: { searchValue: value.searchValue },
			})

			const search = renderer.create(<Search />).toJSON()
			expect(search).toMatchSnapshot()
		})
	})

	it('рендер поля для поиска', () => {
		mockUseAppSelector.mockReturnValue({ filterOptions: { searchValue: '' } })

		render(<Search />)
		const searchInput = screen.getByPlaceholderText('Поиск')
		expect(searchInput).toBeInTheDocument()
	})
	it('отображение текущего значения', () => {
		const searchValue = 'Тест'
		mockUseAppSelector.mockReturnValue({
			filterOptions: { searchValue: searchValue },
		})

		render(<Search />)
		const searchInput = screen.getByPlaceholderText('Поиск')
		expect(searchInput).toHaveValue(searchValue)
	})
	it('передача текущего значения в экшен(setSearchValue)', () => {
		mockUseAppSelector.mockReturnValue({ filterOptions: { searchValue: '' } })

		render(<Search />)
		const searchInput = screen.getByPlaceholderText('Поиск')
		fireEvent.change(searchInput, { target: { value: 'Тест второй' } })
		expect(mockDispatch).toHaveBeenCalledWith(setSearchValue('Тест второй'))
	})
})
