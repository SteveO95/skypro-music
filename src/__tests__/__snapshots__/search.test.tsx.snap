// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Поиск рендер компонента 1`] = `
<div
  className="centerblock__search"
>
  <svg
    className="search__svg"
  >
    <use
      xlinkHref="/img/icon/sprite.svg#icon-search"
    />
  </svg>
  <input
    className="search__text"
    name="search"
    onChange={[Function]}
    placeholder="Поиск"
    type="search"
    value=""
  />
</div>
`;

exports[`Поиск рендер компонента 2`] = `
<div
  className="centerblock__search"
>
  <svg
    className="search__svg"
  >
    <use
      xlinkHref="/img/icon/sprite.svg#icon-search"
    />
  </svg>
  <input
    className="search__text"
    name="search"
    onChange={[Function]}
    placeholder="Поиск"
    type="search"
    value="1"
  />
</div>
`;

exports[`Поиск рендер компонента 3`] = `
<div
  className="centerblock__search"
>
  <svg
    className="search__svg"
  >
    <use
      xlinkHref="/img/icon/sprite.svg#icon-search"
    />
  </svg>
  <input
    className="search__text"
    name="search"
    onChange={[Function]}
    placeholder="Поиск"
    type="search"
    value="один"
  />
</div>
`;
